.github/workflows/workflow.yml

```diff
@@ -0,0 +1,240 @@
+      - name: Build package
+        run: make build
+      - name: Cache release
+        id: restore-release
+        uses: actions/cache@v2
+        with:
+          path: dist
+          key: release-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }}
+
+  test-core:
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      TERM: xterm-256color  # To colorize output of make tasks.
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.12
+      - name: Cache build
+        id: restore-build
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: build-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }}
+      - name: Run openfisca-core tests
+        run:  make test-core
+      - name: Submit coverage to Coveralls
+        run: |
+          pip install coveralls
+          coveralls --service=github
+
+  test-country-template:
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    env:
+      TERM: xterm-256color
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.12
+      - name: Cache build
+        id: restore-build
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: build-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }}
+      - name: Run Country Template tests
+        run: make test-country
+
+  test-extension-template:
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    env:
+      TERM: xterm-256color
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.12
+      - name: Cache build
+        id: restore-build
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: build-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }}
+      - name: Run Extension Template tests
+        run: make test-extension
+
+  check-numpy:
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    env:
+      TERM: xterm-256color
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.12
+      - name: Cache build
+        id: restore-build
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: build-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ github.sha }}
+      - name: Check NumPy typing against latest 3 minor versions
+        run: for i in {1..3}; do VERSION=$(${GITHUB_WORKSPACE}/.github/get-numpy-version.py prev) && pip install numpy==$VERSION && make check-types; done
```

Comment:
I tried using the matrix strategy but it was much more complicated than I anticipated since it is a [dynamic matrix build](https://www.cynkra.com/blog/2020-12-23-dynamic-gha/) which is [problematic](https://github.com/actions/runner/issues/751).

To my understanding it would require, first, creating another job, with a for loop, to get the three Numpy versions to check, store them into a JSON format and then using a matrix strategy (in the job itself) go through the list and install one by one. I was not successful in doing so, but I did not try for very long.

In my opinion, it is a complicated implementation for a simple output that is executed much more elegantly with the existing job and in 30 seconds. If however you think there is a simpler way to do it, or if it's worth the time investment, I'm happy to do it 😉

Reply:
I will merge this PR for the time being. And if need be address this issue in a new one :)

Reply:
IMHO this script is / will be useless once the matrix strategy is in place —if not done already.